storage/store.go

```diff
@@ -1123,6 +1123,10 @@ func (s *Store) addRangeToRangeMap(rng *Range) error {
 func (s *Store) RemoveRange(rng *Range) error {
 	// RemoveGroup needs to access the storage, which in turn needs the
 	// lock. Some care is needed to avoid deadlocks.
+	if !s.stopper.StartTask() {
```

Comment:
we still need to process Raft while draining, it's only right after having drained when Raft no longer needs to be able to make progress. I'll add that check to `processRaft`.